shared/libraries/opcuatms/tests/opcuatms_integration/test_tms_signal.cpp

```diff
@@ -128,19 +157,360 @@ TEST_F(TmsSignalTest, AttrPublic)
     // client side change is reflected on client side (public is not transferred):
     clientSignal.setPublic(false);
     ASSERT_FALSE(clientSignal.getPublic());
-    
+
     clientSignal.setPublic(true);
     ASSERT_TRUE(clientSignal.getPublic());
 }
 
+TEST_F(TmsSignalTest, GetLastValueFloat32)
+{
+    auto daqServerSignal = Signal(NullContext(), nullptr, "id");
+
+    auto serverSignal = TmsServerSignal(daqServerSignal, this->getServer(), ctx, serverContext);
+    auto nodeId = serverSignal.registerOpcUaNode();
+
+    daqServerSignal.setDescriptor(DataDescriptorBuilder().setSampleType(SampleType::Float32).build());
+
+    auto dataPacket = DataPacket(daqServerSignal.getDescriptor(), 5);
+    auto data = static_cast<float*>(dataPacket.getData());
+    data[0] = 0.1f;
+    data[1] = 1.1f;
+    data[2] = 2.1f;
+    data[3] = 3.1f;
+    data[4] = 4.1f;
+
+    daqServerSignal.sendPacket(dataPacket);
+
+    auto clientSignal = TmsClientSignal(NullContext(), nullptr, "sig", clientContext, nodeId);
+
+    ASSERT_FLOAT_EQ(clientSignal.getLastValue(), 4.1f);
+    ASSERT_FLOAT_EQ(daqServerSignal.getLastValue(), 4.1f);
+}
+
+TEST_F(TmsSignalTest, GetLastValueFloat64)
+{
+    auto daqServerSignal = Signal(NullContext(), nullptr, "id");
+
+    auto serverSignal = TmsServerSignal(daqServerSignal, this->getServer(), ctx, serverContext);
+    auto nodeId = serverSignal.registerOpcUaNode();
+
+    daqServerSignal.setDescriptor(DataDescriptorBuilder().setSampleType(SampleType::Float64).build());
+
+    auto dataPacket = DataPacket(daqServerSignal.getDescriptor(), 5);
+    auto data = static_cast<double*>(dataPacket.getData());
+    data[0] = 0.1;
+    data[1] = 1.1;
+    data[2] = 2.1;
+    data[3] = 3.1;
+    data[4] = 4.1;
```

Comment:
This setup is pretty much the same for all LastValue test cases. Could be refactored into a function.